test/unit/Insurance.js

```diff
@@ -1,101 +1,390 @@
 const { expect } = require("chai")
 const { ethers, getNamedAccounts, deployments } = require("hardhat")
 const { deploy } = deployments
+const { smockit } = require("@eth-optimism/smock")
+const { BigNumber } = require("ethers")
+const zeroAddress = "0x0000000000000000000000000000000000000000"
 
 describe("Unit tests: Insurance.sol", function () {
-    before(async function () {
-        // todo
-        // deploy mock tracer
+    let testToken
+    let insurance
+    let mockTracer
+    let accounts
+    let _deployer
+
+    beforeEach(async function () {
+        const { deployer } = await getNamedAccounts()
+        _deployer = deployer
+        // deploy a test token
+        const TestToken = await ethers.getContractFactory("TestToken")
+        testToken = await TestToken.deploy(ethers.utils.parseEther("100000000"))
+        await testToken.deployed()
+
+        // deploy mock tracer and libs
+        let libBalances = await deploy("Balances", {
+            from: deployer,
+            log: true,
+        })
+
+        let libPerpetuals = await deploy("Perpetuals", {
+            from: deployer,
+            log: true,
+        })
+
+        let libPrices = await deploy("Prices", {
+            from: deployer,
+            log: true,
+        })
+
+        // this deploy method is needed for mocking
+        const tracerContractFactory = await ethers.getContractFactory(
+            "TracerPerpetualSwaps",
+            {
+                libraries: {
+                    Balances: libBalances.address,
+                    Perpetuals: libPerpetuals.address,
+                    Prices: libPrices.address,
+                },
+            }
+        )
```

Comment:
These should be in `deploy/FullDeploy.js` now

Reply:
@BenjaminPatch including the mock contract and what not?